### PR TITLE
fix(UI): Disable write access from UI

### PR DESCRIPTION
**Pull request by nikkuma7**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? #1597
> * Did you add or update any new dependencies that are required for your change? No

Issue: 
No use of write access on preferences page, so we removed write access from UI

1) First reviewer login into sw360 project 
2) Go to preferences tab
3) Go down on the same page and write access will not be there.
> Have you implemented any additional tests?

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Review by akapti — CHANGES_REQUESTED** [review]
Tested the changes.
The feature is working fine, but the confusion is:

- write token checkbox is disabled when the property `preferenses.write.access.disable` value is set to `false`.
- and checkbox is enabled when we set the property value to `true`.

Ideally it should work other way round based on name of property key (`preferenses.write.access.disable`).

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/preferences/UserPreferencesPortlet.java [hunk 1]
```diff
@@ -60,7 +60,6 @@ public class UserPreferencesPortlet extends Sw360Portlet {
     private static final Logger log = LogManager.getLogger(UserPreferencesPortlet.class);
     private static final String AUTHORITIES_READ = "AUTHORITIESREAD";
     private static final String AUTHORITIES_WRITE = "AUTHORITIESWRITE";
-
```
Comment (akapti) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/preferences/UserPreferencesPortlet.java:63]: Please don't remove this line.. also at line 188.. Keep this class unchanged.

**Review by akapti — CHANGES_REQUESTED** [review]
Kindly incorporate one minor review comment.

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/preferences/view.jsp [hunk 2]
```diff
@@ -157,9 +157,11 @@
                                                 id="authorities_read" class="form-check-input" />
                                             <label class="form-check-label" for="authorities_read"><liferay-ui:message key="read.access" /></label>
                                             <br>
+                                            <core_rt:if test="${disablewriteaccess}">
                                             <input type="checkbox" name="<portlet:namespace/><%=RestApiToken._Fields.AUTHORITIES%>WRITE"
-                                                id="authorities_write" class="form-check-input" />
-                                            <label class="form-check-label" for="authorities_write"><liferay-ui:message key="write.access" /></label>
+                                                id="authorities_write" class="form-check-input" /> 
+                                            <label class="form-check-label" for="authorities_write"><liferay-ui:message key="write.access" /></label>
```
[the remark this answers is not in the document]
Comment (akapti) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/html/preferences/view.jsp]: still there is whitespace at end of line number `162`.

**Comment by nikkuma7** [comment]
I have modified the code.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/PortalConstants.java [hunk 3]
```diff
@@ -649,6 +649,7 @@ public class PortalConstants {
     // Rest API constants
     public static final UserGroup API_WRITE_ACCESS_USERGROUP;
     public static final Boolean API_TOKEN_ENABLE_GENERATOR;
+    public static final Boolean DISABLE_REST_API_WRITE_ACCESS_TOKEN_IN_PREFERENCES;
```
Comment (akapti) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/PortalConstants.java:652]: please rename this field to `REST_API_WRITE_ACCESS_TOKEN_IN_PREFERENCES`... in line with property key.

**Review by akapti — CHANGES_REQUESTED** [review]
One minor review comment.
Please squash multiple commits.

**Review by akapti — APPROVED** [review]
Looks good to me.

**Review by akapti — CHANGES_REQUESTED** [review]
Getting following while testing the PR.

```
2022-09-05 08:20:04.747 ERROR [http-nio-8080-exec-3][PortletRequestDispatcherImpl:304] Unable to dispatch request: PWC6033: Error in Javac compilation for JSP__PWC6197: An error occurred at line: 33 in the jsp file: /html/preferences/view.jsp_PWC6199: Generated servlet error:_cannot find symbol_  symbol:   variable DISABLE_REST_API_WRITE_ACCESS_TOKEN_IN_PREFERENCES_  location: class org.eclipse.sw360.portal.common.PortalConstants__PWC6199: Generated servlet error:_view_jsp.java uses or overrides a deprecated API.__PWC6199: Generated servlet error:_Recompile with -Xlint:deprecation for details.__PWC6199: Generated servlet error:_view_jsp.java uses unchecked or unsafe operations.__PWC6199: Generated servlet error:_Recompile with -Xlint:unchecked for details.__ [Sanitized]
2022-09-05 08:20:04.749 ERROR [http-nio-8080-exec-3][PortletServlet:119] null
com.liferay.portal.kernel.log.LogSanitizerException: javax.portlet.PortletException: org.apache.jasper.JasperException: PWC6033: Error in Javac compilation for JSP__PWC6197: An error occurred at line: 33 in the jsp file: /html/preferences/view.jsp_PWC6199: Generated servlet error:_cannot find symbol_  symbol:   variable DISABLE_REST_API_WRITE_ACCESS_TOKEN_IN_PREFERENCES_  location: class org.eclipse.sw360.portal.common.PortalConstants__PWC6199: Generated servlet error:_view_jsp.java uses or overrides a deprecated API.__PWC6199: Generated servlet error:_Recompile with -Xlint:deprecation for details.__PWC6199: Generated servlet error:_view_jsp.java uses unchecked or unsafe operations.__PWC6199: Generated servlet error:_Recompile with -Xlint:unchecked for details.__ [Sanitized]
```

Please fix it.
screenshot: 
image

**Review by akapti — APPROVED** [review]
code looks good. working as expected.

**Pull request merged**